### PR TITLE
added hot module replacement port

### PR DESCRIPTION
**Pull request by RizkyRajitha**
# ↪️ Pull Request

added  hot module replacement port cli argument 
issue #5150 

## 💻 Examples

`parcel --port 8080 --hmr-port 8080  src/index.js`

### CODE REVIEW

**Comment by mischnic** [comment]
This doesn't do anything at the moment because the value isn't used anywhere.

If you run `parcel src/index.html --hmr-port 4444` in the "html" example, this should happen:

Opening `http://localhost:1234/` should serve the page and if you check the "Network" Tab of the browser devtools, there should be a connection made to `ws://localhost:4444/`

---

https://github.com/parcel-bundler/parcel/blob/52f365668013a08b8afa03e549934a6137eb9c4d/packages/core/parcel/src/cli.js#L345-L347

The argument to `--hmr-port` is available as `command.hmrPort`

**Comment by RizkyRajitha** [comment]
> This doesn't do anything at the moment because the value isn't used anywhere.
> 
> If you run `parcel src/index.html --hmr-port 4444` in the "html" example, this should happen:
> 
> The HTTP server starts on `http://localhost:1234/` and if you check the "Network" tab of the browser devtools, there should be a connection made to `ws://localhost:4444/` instead of `1234`
> 
> Bildschirmfoto 2020-09-27 um 18 14 36
> 
> https://github.com/parcel-bundler/parcel/blob/52f365668013a08b8afa03e549934a6137eb9c4d/packages/core/parcel/src/cli.js#L345-L347
> 
> The argument to `--hmr-port` is available as `command.hmrPort`

i will take a look and let you know

thanks

**Comment by RizkyRajitha** [comment]
hi 
added changes with new commit 

now working as expected 

### no specified ports

no hmr port

dev1

### with both ports




dev2


### with only hmr port

hmrtest1

dev3

**Comment by mischnic** [comment]
`command.hmrPort` is a string and `hot.port` is expected to be a number:

https://github.com/parcel-bundler/parcel/blob/b2da8a221b9289b626cbef2742894101ca620677/packages/core/types/index.js#L51

Maybe this gets normalized along the way anyway, but we should still do `parceInt` or `Number()` somewhere in cli.js:
https://github.com/parcel-bundler/parcel/blob/b2da8a221b9289b626cbef2742894101ca620677/packages/core/parcel/src/cli.js#L92

**Comment by RizkyRajitha** [comment]
> `command.hmrPort` is a string and `hot.port` is expected to be a number:
> 
> https://github.com/parcel-bundler/parcel/blob/b2da8a221b9289b626cbef2742894101ca620677/packages/core/types/index.js#L51
> 
> Maybe this gets normalized along the way anyway, but we should still do `parceInt` or `Number()` somewhere in cli.js:
> https://github.com/parcel-bundler/parcel/blob/b2da8a221b9289b626cbef2742894101ca620677/packages/core/parcel/src/cli.js#L92

`let hmrport = command.hmrPort ? Number(command.hmrPort) : port;`

will this type of conversion is ok ?

thanks

**Comment by RizkyRajitha** [comment]
hi  @mischnic 
i am getting this error after the merge with v2 

merge err 1

goes away if i add the port

merge e1

**Comment by mischnic** [comment]
> i am getting this error after the merge with v2

Delete your cache folder. Since a recent PR, more more information is stored in the cache and that data is missing in your preexisting cache folder.

**Comment by RizkyRajitha** [comment]
hi @mischnic 
i add a commit with `Number()` conversion 
thanks

File: packages/examples/html/package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "license": "MIT",
   "private": true,
   "scripts": {
-    "demo": "parcel build src/index.html"
+    "demo": "parcel build src/index.html",
+    "hmrtest": "parcel src/index.html --hmr-port 4444"
```
Comment (mischnic) [inline packages/examples/html/package.json:8]:
```suggestion
    "demo": "parcel build src/index.html"
```

File: packages/examples/simple/package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "private": true,
   "scripts": {
     "demo": "PARCEL_WORKERS=0 parcel src/index.js",
+    "hmr-demo": "parcel --port 8080 --hmr-port 8080  src/index.js ",
```
Comment (mischnic) [inline packages/examples/simple/package.json:8]:
```suggestion
```

**Review by mischnic — APPROVED** [review]

**Comment by RizkyRajitha** [comment]
thank you for the generous help and support @mischnic .

**Comment by RizkyRajitha** [comment]
hi @mischnic ,
why some checks are failing . is there something i can do to solve that.
thank you

**Comment by mischnic** [comment]
No, I'll make sure they pass now and the merge 👍

**Comment by RizkyRajitha** [comment]
> No, I'll make sure they pass now and the merge 👍

@mischnic 
Thank you
looking forward to contribute again .

**Pull request merged**